tsfile/src/main/java/org/apache/iotdb/tsfile/utils/ReadWriteForEncodingUtils.java

```diff
@@ -98,14 +98,24 @@ public static int getLongMaxBitWidth(List<Long> list) {
   public static int readUnsignedVarInt(InputStream in) throws IOException {
     int value = 0;
     int i = 0;
-    int b;
-    while (((b = in.read()) & 0x80) != 0) {
+    int b = in.read();
+    while (b != -1 && (b & 0x80) != 0) {
       value |= (b & 0x7F) << i;
       i += 7;
+      b = in.read();
     }
     return value | (b << i);
   }
 
+  public static int readVarInt(InputStream in) throws IOException {
```

Comment:
correct? how about -1?

Reply:
Yes, it is right, i will add some ut for this class